add docs for the Navigation API (#21960)
mdn/content: files/en-us/web/api/navigation/reload/index.md

{{domxref("Navigation")}} interface reloads the current URL, updating any provided state in the history entries list.

## Syntax

```js-nolint
navigate(options)
```

### Parameters

- `options` {{optional_inline}}
  - : An options object containing the following properties:
    - `state`
      - : Developer-defined information to be stored in the associated {{domxref("NavigationHistoryEntry")}} once the navigation is complete, retrievable via {{domxref("NavigationHistoryEntry.getState", "getState()")}}. This can be any data type. You might, for example, wish to store a page visit count for analytics purposes, or store UI state details so the view can be shown exactly as the user last left it.
    - `info`
      - : Developer-defined information to be passed along to the {{domxref("Navigation/navigate_event", "navigate")}} event, made available in {{domxref("NavigateEvent.info")}}. This can be any data type. You might, for example, wish to display newly-navigated content with a different animation depending on how it was navigated to (swipe left, swipe right, or go home). A string indicating which animation to use could be passed in as `info`.

### Return value

> An object containing two promises — `{ committed, finished }`. This allows the invoking function to wait on taking further action until:
> 
> - `committed` fulfills, meaning that the visible URL has changed and a new {{domxref("NavigationHistoryEntry")}} has been created.
> - `finished` fulfills, meaning that all promises returned by the `intercept()` handler are fulfilled. This is equivalent to the {{domxref("NavigationTransition.finished")}} promise fulfilling, when the {{domxref("Navigation/navigatesuccess_event", "navigatesuccess")}} event fires.
> - either one of the above promises rejects, meaning that the navigation has failed for some reason.

Reviewer: ```suggestion
An object with the following properties:

- `committed`
  - : A {{jsxref("Promise")}} which will fulfill when the visible URL has changed and a new {{domxref("NavigationHistoryEntry")}} has been created.
- `finished`
  - : A {{jsxref("Promise")}} which will fulfill when all promises returned by the `intercept()` handler are fulfilled. This is equivalent to the {{domxref("NavigationTransition.finished")}} promise fulfilling, when the {{domxref("Navigation/navigatesuccess_event", "navigatesuccess")}} event fires.
 
Either one of these promises rejects if the navigation has failed for some reason.
```

### Exceptions

None.

## Examples

> Usage of info and state

Reviewer: ```suggestion
### Using info and state
```
